config/opal_check_libnl.m4

```diff
@@ -0,0 +1,262 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2015-2016 Research Organization for Information Science
+dnl                         and Technology (RIST). All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+dnl
+dnl More libnl v1/v3 sadness: the two versions are not compatible
+dnl and will not work correctly if simultaneously linked into the
+dnl same applications.  Unfortunately, they *will* link into the
+dnl same image!  On platforms like SLES 12, libibverbs depends on
+dnl libnl-3.so.200 and friends, while a naive implementation of
+dnl our configure logic would link libnl.so.1 to libdaplusnic,
+dnl resulting in both versions in the dependency map at the same
+dnl time.
```

Comment:
@ggouaillardet I think you might want to edit this comment (vs. blind copy-n-paste).  E.g., the references here to DAPL aren't relevant.  😄 

What I'm looking for here is both a description of the problem (which this first paragraph essentially does), and also at least some description of the overall strategy we're doing here in this M4 file to try to avoid the problem (e.g., after each OPAL_CHECK_PACKAGE, check to see if any of the dependent libraries include libnl/libnl3, detect conflicts, ...etc.).